DependencyInjection/LiipMonitorExtension.php

```diff
@@ -129,4 +152,192 @@ private function setParameters(ContainerBuilder $container, $checkName, $group,
+            $serviceConfiguration->addMethodCall('setName', [ $configuration->getName() ]);
+        }
+
+        if ($configuration->getMigrationsDirectory()) {
+            $directory        = $configuration->getMigrationsDirectory();
+            $pathPlaceholders = array('kernel.root_dir', 'kernel.cache_dir', 'kernel.logs_dir');
+            foreach ($pathPlaceholders as $parameter) {
+                $kernelDir = realpath($container->getParameter($parameter));
+                if (strpos(realpath($directory), $kernelDir) === 0) {
+                    $directory = str_replace($kernelDir, "%{$parameter}%", $directory);
+                    break;
+                }
+            }
+
+
+            $serviceConfiguration->addMethodCall(
+                'setMigrationsDirectory',
+                [ $directory ]
+            );
+        }
+
+        /** @var AbstractFileConfiguration $diff */
+        $versions = $this->getPredefinedMigrations($configuration, $connection);
+        if ($versions) {
+            $serviceConfiguration->addMethodCall('registerMigrations', [ $versions ]);
+        }
+
+        $serviceConfiguration->addMethodCall('configure', []);
+
+        if ($configuration->areMigrationsOrganizedByYear()) {
+            $serviceConfiguration->addMethodCall('setMigrationsAreOrganizedByYear', [ true ]);
+
+            return $serviceConfiguration;
+        } elseif ($configuration->areMigrationsOrganizedByYearAndMonth()) {
+            $serviceConfiguration->addMethodCall('setMigrationsAreOrganizedByYearAndMonth', [ true ]);
+
+            return $serviceConfiguration;
+        }
+
+        return $serviceConfiguration;
+    }
+
+    /**
+     * Creates in-memory migration configuration for setting up container service
+     *
+     * @param ContainerBuilder $container  The container
+     * @param Connection       $connection Fake connection
+     * @param string           $filename   Migrations configuration file
+     *
+     * @return AbstractFileConfiguration
+     */
+    private function createTemporaryConfiguration(ContainerBuilder $container, Connection $connection, $filename)
+    {
+        // -------
+        // This part must be in sync with Doctrine\DBAL\Migrations\Tools\Console\Helper\ConfigurationHelper::loadConfig
+        $map = [
+            'xml'  => '\XmlConfiguration',
+            'yaml' => '\YamlConfiguration',
+            'yml'  => '\YamlConfiguration',
+            'php'  => '\ArrayConfiguration',
+            'json' => '\JsonConfiguration',
+        ];
+        // --------
+
+        $filename = $container->getParameterBag()->resolveValue($filename);
+        $info     = pathinfo($filename);
+        // check we can support this file type
+        if (empty($map[ $info[ 'extension' ] ])) {
+            throw new \InvalidArgumentException('Given config file type is not supported');
+        }
+
+        $class = 'Doctrine\DBAL\Migrations\Configuration';
+        $class .= $map[ $info[ 'extension' ] ];
```

Comment:
I guess you kept this part here instead of in the DI extension because there could be multiple different files in different formats?